report/report.md

```diff
@@ -0,0 +1,898 @@
+---
+title: Investigating IVC with Halo2
+date: \today
+author: 
+  - Rasmus Kirk Jakobsen - 201907084
+geometry: margin=2cm
+---
+
+\tableofcontents
+\newpage
+
+# Introduction
+
+Halo2, can be broken down into the following components:
+
+- **Plonk**: A general-purpose, potentially zero-knowledge, proof scheme.
+- **$\PCDL$**: A Polynomial Commitment Scheme in the Discrete Log setting.
+- **$\ASDL$**: An Accumulation Scheme in the Discrete Log setting.
+- **Pasta**: A Cycle of Elliptic Curves, namely **Pa**llas and Ve**sta**.
+
+This project is focused on the components of $\PCDL$ and $\ASDL$. I used the
+[2020 paper](https://example.com) _"Proof-Carrying Data from Accumulation
+Schemes"_ as a reference. The project covers both the theoretical aspects
+of the scheme described in this document along with a rust implementation,
+both of which can be found in the project's [repository](https://example.com).
+
+## Prerequisites
+
+Basic knowledge on elliptic curves, groups, interactive arguments are
+assumed in the following text. There is also a heavy reliance on the Inner
+Product Proof from the Bulletproofs protocol, see the following resources
+on bulletproofs if need be:
+
+- [Section 3 in the original Bulletproofs paper](https://eprint.iacr.org/2017/1066.pdf#section.3)
+- [From Zero (Knowledge) to Bulletproofs writeup](https://github.com/AdamISZ/from0k2bp/blob/master/from0k2bp.pdf)
+- [Rust Dalek Bulletproofs implementation notes](https://doc-internal.dalek.rs/develop/bulletproofs/notes/inner_product_proof/)
+- [Section 4.1 of my bachelors thesis](https://rasmuskirk.com/documents/high-assurance-cryptography-implementing-bulletproofs-in-hacspec.pdf#subsection.4.1)
+
+## Background and Motivation
+
+### Proof Systems
+
+In an Interactive Proof System we have two Interactive Turing machines
+the computationally unbounded Prover, P, and the polynomally time bounded
+Verifier, V. The Prover tries to convince the Verifier of a statement $x \in L$
+language $L \subset \mathbb{B}^*$ in NP. The following properties must be true:
+
+- Completeness: $\forall P \in ITM, x\in L \implies Pr[V_{out} = \bot] \leq \epsilon(x)$
+
+  For all honest provers, P, where $x$ is true, the probability that the
+  verifier remains unconvinced is negligible in the length of $x$.
+
+- Soundness: $\forall P^* \in ITM, x \notin L \implies Pr[V_{out} = \top] \leq \epsilon(x)$
+
+  For all provers, honest or otherwise, $P^*$, that tries to convince the
+  verifier of a claim, $x$, that is not true, the probability that the
+  verifier will be convinced is negligible in the length of $x$.
+
+An Interactive Argument is very similar, but the honest and malicious prover
+is now polynomially bounded and receives a Private Auxilliary Input, for
+example a witness:
```

Comment:
I suggest removing "Private Auxiliary Input" and using "witness" everywhere. The completeness property should then be stated as something like this: for all PPT P that holds w so that (x,w) \in R, it holds that Pr[V_{out} = \bot] \leq \epsilon(x)

Reply:
I had the same annoyance, but was actually afraid whether "Private Auxiliary Input" necessarily meant "witness". I can assume they are always interchangeable?